### DISCUSSION TITLE
Dielectric Material giving black ring inside the sphere

### DISCUSSION THREAD

**Discussion by Shivraj1906** [RayTracing/raytracing.github.io#1035] [Debugging]
I am following the book and writing the code in Java. Everything worked fine until I implemented the Dielectric material.
It is giving the black ring inside the sphere.

I guess the problem is with how I am calculating the normals. But I can't figure it out.

![image](https://user-images.githubusercontent.com/57698531/179181463-e9d9410b-af01-4647-a4f8-a2846bda1f6a.png)

Link to my repo:
https://github.com/Shivraj1906/raytracer.git

**Comment by code-monkey-101** [comment]
The hollow glass sphere should have a radius of -0.4, not -0.5. Also, you are missing the outer layer of the sphere (the second sphere at -1, 0, -1).

![image](https://user-images.githubusercontent.com/32152103/179384603-cf6f7bef-d549-4bb9-8844-88942b2bf62d.png)
![image](https://user-images.githubusercontent.com/32152103/179384611-db85d746-2ba6-4db5-bad3-0a378d65a9f2.png)

**Comment by code-monkey-101** [comment]
Interestingly, you are not the only one with this exact problem. Wondering if there is a buggy version of the book somewhere.

See also https://github.com/RayTracing/raytracing.github.io/discussions/513

**Reply by Shivraj1906** [reply]
That minus radius was just typo there, even after that it is giving the black ring.  I will try to debug the code.

Thanks for your help.

**Reply by code-monkey-101** [reply]
You forgot to copy frontFace in your HitableList.hit method

![image](https://user-images.githubusercontent.com/32152103/179404818-409168bf-f8e7-49ef-8dca-23f40e369b1a.png)

**Reply by Shivraj1906** [reply]
Thanks mate, a lesson to be learned here.
